inc/Composer/Command.php

```diff
@@ -3,32 +3,131 @@
 namespace HM\Platform\LocalServer\Composer;
 
 use Composer\Command\BaseCommand;
+use Symfony\Component\Console\Input\InputOption;
+use Symfony\Component\Console\Input\InputArgument;
 use Symfony\Component\Console\Input\InputInterface;
 use Symfony\Component\Console\Output\OutputInterface;
 use Symfony\Component\Process\Process;
 
 class Command extends BaseCommand {
 	protected function configure() {
-		$this->setName( 'local-server' );
+		$this->setName( 'local-server' )
+			->setDescription( 'Local server.' )
+			->setDefinition( [
+				new InputArgument( 'subcommand', null, 'start, stop, cli, status. logs.' ),
+				new InputArgument( 'options', InputArgument::IS_ARRAY ),
+			] )
+			->setHelp(
+				<<<EOT
+Run the local development server.
+
+To start the local development server:
+	start
+Stop the local development server:
+	stop
+View status of the local development server:
+	status
+Run WP CLI command:
+	cli -- <command>              eg: cli -- post list --debug
+View the logs
+	logs <service>                <service> can be php, nginx, db, s3, elasticsearch, xray
+EOT
+			);
+	}
+
+	public function isProxyCommand() {
+		return true;
 	}
 
 	protected function execute( InputInterface $input, OutputInterface $output ) {
+		$subcommand = $input->getArgument( 'subcommand' );
+
+		if ( $subcommand === 'start' ) {
+			return $this->start( $input, $output );
+		} elseif ( $subcommand === 'stop' ) {
+			return $this->stop( $input, $output );
+		} elseif ( $subcommand === 'cli' ) {
+			return $this->cli( $input, $output );
+		} elseif ( $subcommand === 'status' ) {
+			return $this->status( $input, $output );
+		} elseif ( $subcommand === 'logs' ) {
+			return $this->logs( $input, $output );
+		}
+	}
+
+	protected function start( InputInterface $input, OutputInterface $output ) {
 		$output->writeln( 'Starting...' );
 
-		$proxy = new Process( 'docker-compose -f proxy.yml up', 'vendor/humanmade/local-server/docker' );
-		$proxy->setTimeout( 0 );
-		$proxy->start();
+		$proxy = new Process( 'docker-compose -f proxy.yml up -d', 'vendor/humanmade/local-server/docker' );
+		$proxy->run();
 
-		$compose = new Process( 'docker-compose up', 'vendor/humanmade/local-server/docker', [
+		$compose = new Process( 'docker-compose up -d', 'vendor/humanmade/local-server/docker', [
 			'VOLUME' => getcwd(),
 			'COMPOSE_PROJECT_NAME' => basename( getcwd() ),
 		] );
 		$compose->setTimeout( 0 );
+		$compose->run( function ( $type, $buffer ) {
+			echo $buffer;
+		} );
+
+		$site_url = 'https://' . basename( getcwd() ) . '.altis.dev/';
+		$output->writeln( 'Startup completed.' );
+		$output->writeln( 'To access your site visit: ' . $site_url );
+	}
+
+	protected function stop( InputInterface $input, OutputInterface $output ) {
+		$output->writeln( 'Stopping...' );
+
+		$proxy = new Process( 'docker-compose down', 'vendor/humanmade/local-server/docker' );
```

Comment:
Does this exit with an error if it's already stopped?

Reply:
I think it's docker-compose is happy with it